platform/features/conductor-v2/conductor/src/ui/MctConductorAxis.js

```diff
@@ -44,16 +44,27 @@ define(
                             .append('svg:svg')
                             .attr('width', '100%')
                             .attr('height', height);
-                var xScale = d3.scaleUtc();
                 var xAxis = d3.axisTop();
                 // draw x axis with labels and move to the bottom of the chart area
                 var axisElement = vis.append("g")
                     .attr("transform", "translate(0," + (height - padding) + ")");
 
-                function setScale(start, end) {
+                function setScale() {
+                    var xScale = undefined;
                     var width = target.offsetWidth;
-                    xScale.domain([new Date(start), new Date(end)])
-                        .range([padding, width - padding * 2]);
+                    var timeSystem = conductor.timeSystem();
+                    var bounds = conductor.bounds();
+
+                    //If the time can be represented as a date, then use a
+                    // date scale. Otherwise use a standard linear scale
+                    if (timeSystem.toDate(bounds.start) !== false) {
```

Comment:
You wrote a comment to explain unclear code.  Would be better to write code that is clear.

If the time system is utc based, then the only proper implementation of `toDate` is `new Date(val)`.  When describing the timesystem, describe what it is.

I suggest:
```javascript
if (timeSystem.isUTCBased()) { 
  xScale = d3.scaleUtc()
    .domain([new Date(start), new Date(end)]); 
} else { 
  xScale = d3.scaleLinear()
    .range([start, end]);
}
```

Using the return value of `toDate` for the check may be clever but is not clear.